### PR TITLE
[editorial] Setup Prettier and run it on some files

### PR DESCRIPTION
**Pull request by chalin**
- A first step towards #189.
- We can then progressively add files, and/or tune Prettier options.

/cc @jsuereth @joaopgrassi

### CODE REVIEW

**Comment by joaopgrassi** [comment]
Can we run it on one of the other files (maybe one of the largest ones?) to see what kind of changes would happen before we decide to start using it? I'm generally in favor for adding such things to take away debates about style :)

**Comment by chalin** [comment]
Sure @joaopgrassi. I've added all `/docs/cloud*` files to the list of processed files. Does that help?

Note the required use of `<!-- prettier-ignore-start/end -->` before autogenerated semconv tables.

File: package.json [hunk 1]
```diff
@@ -2,6 +2,10 @@
   "devDependencies": {
     "markdown-link-check": "3.10.3",
     "markdown-toc": "^1.2.0",
-    "markdownlint-cli": "0.31.0"
+    "markdownlint-cli": "0.31.0",
+    "prettier": "^3.0.0"
+  },
+  "prettier": {
+    "proseWrap": "preserve"
```
Comment (chalin) [inline package.json:9]: Note that eventually we can to change this option to `always`. That makes for a better experience viewing PR diffs in GitHub. For now, to minimize changes, we should keep `preserve`.

**Comment by chalin** [comment]
FYI, if you select the "rich diff" view, you'll see that there are no changes for most files, e.g.:

> image

File: docs/cloudevents/cloudevents-spans.md [hunk 2]
```diff
@@ -39,7 +40,7 @@ document.
 A CloudEvent can be sent directly from producer to consumer.
 For such a scenario, the traditional parent-child trace model works well.
 However, CloudEvents are also used in distributed systems where an event
-can go through many [hops](https://en.wikipedia.org/wiki/Hop_(networking))
+can go through many [hops](<https://en.wikipedia.org/wiki/Hop_(networking)>)
```
[the remark this answers is not in the document]
Comment (chalin) [reply docs/cloudevents/cloudevents-spans.md:43]: No, only in cases where a URL contains characters that _might_confuse markdown processors: in this case, the presence of parentheses is that triggers this adjustment.

**Comment by joaopgrassi** [comment]
Do we need to maybe add instructions on how to "invoke" prettier to apply changes/fixes locally? Thinking on how people should address their changes after this is merged. Or do you plan to do this as a follow up? Also - do we want to add checks as a GitHub action?

**Comment by chalin** [comment]
> Do we need to maybe add instructions on how to "invoke" prettier to apply changes/fixes locally?

Done. I've added scripts and updated the CONTRIBUTING instructions.

(I haven't yet added a CI check.)

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -61,7 +62,7 @@ Semantic conventions have some autogenerated components and additionally can do
 automatic style/spell correction. You can run all of this via:
 
 ```bash
-make fix
+npm run fix
```
Comment (jsuereth) [inline CONTRIBUTING.md:65]: Can we keep `make` as the entry point, and just have it call `npm` ?

Comment (chalin) [reply CONTRIBUTING.md]: Certainly. Done, PTAL.

**Review by arminru — APPROVED** [review]

**Pull request merged**